packages/block-library/src/template-part/edit/placeholder/index.js

```diff
@@ -15,26 +15,34 @@ import { store as coreStore } from '@wordpress/core-data';
 import TemplatePartSelection from '../selection';
 
 export default function TemplatePartPlaceholder( {
+	area,
 	setAttributes,
 	innerBlocks,
 } ) {
 	const { saveEntityRecord } = useDispatch( coreStore );
 	const onCreate = useCallback( async () => {
 		const title = __( 'Untitled Template Part' );
+		const record = {
+			title,
+			slug: 'template-part',
+			content: serialize( innerBlocks ),
+		};
+		// If we have `area` set from block attributes, means an exposed
+		// block variation was inserted. So add this prop to the template
+		// part entity on creation. Afterwards remove `area` value from
+		// block attributes.
+		if ( [ 'header', 'footer' ].includes( area ) ) record.area = area;
```

Comment:
There are more values available and they are filterable on the server:
https://github.com/WordPress/gutenberg/blob/a5395f8e4a88b59816af1753a558c0e1208430bb/lib/full-site-editing/template-parts.php#L171-L183

Reply:
Yeah, but we are exposing only the above two in the Inserter (header, footer). When we decide to add the `Sidebar` we will include this as well. Regarding the `uncategorized`, is the default if not provided so we don't need anything explicit provided there for the base Template part block.

Reply:
> Yeah, but we are exposing only the above two in the Inserter (header, footer). When we decide to add the `Sidebar` we will include this as well. 

How do you plan to ensure it happens when someone changes the list of block variations? I don't think those two should know of each other. You also need to keep in mind that block variations can be filtered.

Reply:
I got what you mean now, thanks for explaining. I removed the check since it is handled on the server and it defaults to `uncategorized` if the provided value is not allowed.